core/domain/exp_services.py

```diff
@@ -641,64 +765,155 @@ def apply_change_list(
+                written_translations_as_needing_update = cast(
+                    exp_domain.MarkWrittenTranslationsAsNeedingUpdateCmd,
+                    change
+                )
                 exploration.states[
-                    change.state_name
-                ].mark_written_translations_as_needing_update(change.content_id)
+                    written_translations_as_needing_update.state_name
+                ].mark_written_translations_as_needing_update(
+                    written_translations_as_needing_update.content_id
+                )
             elif change.cmd == exp_domain.CMD_EDIT_EXPLORATION_PROPERTY:
-                if change.property_name == 'title':
-                    exploration.update_title(change.new_value)
-                elif change.property_name == 'category':
-                    exploration.update_category(change.new_value)
-                elif change.property_name == 'objective':
-                    exploration.update_objective(change.new_value)
-                elif change.property_name == 'language_code':
-                    exploration.update_language_code(change.new_value)
-                elif change.property_name == 'tags':
-                    # Ruling out the possibility of any other type for mypy
-                    # type checking.
-                    assert isinstance(change.new_value, list)
-                    exploration.update_tags(change.new_value)
-                elif change.property_name == 'blurb':
-                    exploration.update_blurb(change.new_value)
-                elif change.property_name == 'author_notes':
-                    exploration.update_author_notes(change.new_value)
-                elif change.property_name == 'param_specs':
-                    # Ruling out the possibility of any other type for mypy
-                    # type checking.
-                    assert isinstance(change.new_value, dict)
-                    exploration.update_param_specs(change.new_value)
-                elif change.property_name == 'param_changes':
-                    # Ruling out the possibility of any other type for mypy
-                    # type checking.
-                    assert isinstance(change.new_value, list)
+                # Here we use cast because we are narrowing down the type from
+                # ExplorationChange to a specific change command.
+                edit_exp_property_cmd = cast(
+                    exp_domain.EditExplorationPropertyCmd,
+                    change
+                )
+                if edit_exp_property_cmd.property_name == 'title':
+                    # Here we use cast because in this 'if clause' we are
+                    # updating the title of a exploration which can only be
+                    # of type string. So, to rule out all other property types
+                    # for MyPy type checking, we used cast here.
+                    title = cast(str, edit_exp_property_cmd.new_value)
```

Comment:
Similar notes, cast the edit_exp_property_cmd to the relevant subtype. I actually think you can then shorten the comment, e.g. "This 'if' condition forces edit_exp_property_cmd to have type XXX."

Btw, can you use literals in the subtype definitions (so, property_name is typed as a literal rather than an arbitrary string)?

Reply:
using `This 'if' condition forces edit_exp_property_cmd to have type XXX` similar comment to reduce the comment length.

> Btw, can you use literals in the subtype definitions -- done